test/pjrt/test_ddp.py

```diff
@@ -18,39 +17,29 @@
 
 FLAGS = args_parse.parse_common_options()
 
-ddp_parameters = [('torch_xla', pjrt.DistributedDataParallel)]
-if pjrt.device_type() == 'TPU' and tpu.version() >= 4:
-  ddp_parameters.append(('torch', torch.nn.parallel.DistributedDataParallel))
-
 
 class TestPjRtDistributedDataParallel(parameterized.TestCase):
 
-  def setUp(self):
-    os.environ['PJRT_INIT_TORCH_DISTRIBUTED'] = '1'
-
   @staticmethod
-  def _ddp_init(ddp):
+  def _ddp_init(index: int = ...):
+    dist.init_process_group('xla', init_method='pjrt://')
     device = xm.xla_device()
     model = nn.Linear(10, 10).to(device)
-    ddp_model = ddp(model)
-
-  @parameterized.named_parameters(*ddp_parameters)
-  def test_ddp_init(self, ddp: type):
-    pjrt._run_multiprocess(self._ddp_init, ddp)
+    ddp_model = DDP(model)
 
-  def test_ddp_init_no_distributed(self):
-    with self.assertRaises(RuntimeError):
-      with mock.patch.dict(os.environ, {'PJRT_INIT_TORCH_DISTRIBUTED': '0'}):
-        pjrt._run_multiprocess(self._ddp_init, pjrt.DistributedDataParallel)
+  def test_ddp_init(self):
+    pjrt._run_multiprocess(self._ddp_init)
 
-  @parameterized.named_parameters(*ddp_parameters)
-  def test_ddp_correctness(self, ddp: type):
-    pjrt._run_multiprocess(util.ddp_correctness, ddp=ddp, debug=FLAGS.debug)
+  def test_ddp_init_threaded(self):
+    pjrt.spawn_threads(self._ddp_init)
 
-  @parameterized.named_parameters(*ddp_parameters)
-  def test_ddp_correctness_large_net(self, ddp: type):
+  @parameterized.named_parameters(('small_net', False), ('large_net', True))
+  def test_ddp_correctness(self, use_large_net: bool):
     pjrt._run_multiprocess(
-        util.ddp_correctness, ddp=ddp, use_large_net=True, debug=FLAGS.debug)
+        util.ddp_correctness,
+        init_method='pjrt://',
```

Comment:
Wonder if we want to parameterized the init_method with env as well?

Reply:
Good idea. Added another test that skips for TPU <= v3, since `env://` doesn't work nicely with multithreading.